### PR TITLE
Fix reported bug in delete command parse and update PPP 

### LINKED ISSUE #179: [PE-D][Tester E] Error message is not consistent for invalid indices
The error messages are not consistent for different invalid indices. For example, there is no one at person index 0 or 1000 therefore the error messages should be the same. 

After running `delete p 0` 

After running `delete p 1000`

After running `delete p -999`

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: nreHieW/ped#13

### LINKED ISSUE #181: [PE-D][Tester E] Delete internship output message consistency
The output success message for deleting an internship has the wrong index. In the example below, I deleted the internship at index 1 but the output says I deleted the internship at index 0 instead. Similar to other issues, the UI is not updated to reflect this change. 

This is before:

This is after `delete i c/5 i/1`

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: nreHieW/ped#14

### PR DESCRIPTION
**Pull request by rngzhi**
close #181 
close #179

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/DeleteCommandParser.java [hunk 1]
```diff
@@ -45,60 +45,90 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
...
-                }
-            } catch (ParseException e) {
-                // If index provided is not an integer
-                throw new ParseException(
-                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteInternshipCommand.MESSAGE_USAGE));
-            }
+        String[] commandParts = trimmedArgs.split("\\s+");
 
+        if (isDeleteInternshipCommand(commandParts)) {
+            return createDeleteInternshipCommand(commandParts);
```
Comment (Tim-Siu) [inline src/main/java/seedu/address/logic/parser/DeleteCommandParser.java:56]: This makes the code much more readable. Great!

File: src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java [hunk 2]
```diff
@@ -19,8 +19,8 @@ public static void assertParseSuccess(Parser<? extends Command> parser, String u
         try {
             Command command = parser.parse(userInput);
             assertEquals(expectedCommand, command);
-        } catch (ParseException pe) {
-            throw new IllegalArgumentException("Invalid userInput.", pe);
+        } catch (ParseException e) {
+            throw new IllegalArgumentException("Invalid userInput.", e);
```
Comment (Tim-Siu) [inline src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java:23]: Why did you changed `pe` to `e` here, but not for the method below?

**Review by Tim-Siu — APPROVED** [review]

File: src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java (hunk 2 above)
Comment (rngzhi) [reply src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java:23]: Previously I adjust the catch clause to also handle CommandExcption, thus the variable for exception should be more general. But later I realized that Parser classes should only throw ParseException.

Comment (rngzhi) [reply src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java:23]: There's no difference here.

**Pull request merged**